app/components/form/fields/DiskSizeField.tsx

```diff
@@ -17,12 +22,19 @@ interface DiskSizeProps<
   TName extends FieldPath<TFieldValues>,
 > extends TextFieldProps<TFieldValues, TName> {
   minSize?: number
+  validate?(diskSizeGiB: number): ValidateResult
```

Comment:
This is more of a FYI about React Hook form, not a suggestion, because I can't bring myself to suggest something so much worse than what you have, and I don't think it's really more typesafe in practice. But if you hover on the validate prop, you can see it is using some helpers from RHF to look up the value of the field using the field name.

![Screenshot 2023-12-06 at 5 00 41 PM](https://github.com/oxidecomputer/console/assets/3612203/49ef3223-a47c-46c2-97fd-45590d2682d2)

If you wanted to match that type on the prop, you could do this:

```ts
  validate?: Validate<PathValue<TFieldValues, TName>, TFieldValues>
```

This produces the same result as what you have (the argument is a number, of course) except it also has a second argument, which is all the form values together:

<img width="1055" alt="image" src="https://github.com/oxidecomputer/console/assets/3612203/cb0fb407-c817-49b1-b6af-79bf0d95eb8e">

This could be handy in theory because you could validate one field value based on the value of another. But we're not doing that — well, we're doing it, but we're doing it at the call site so this field doesn't have to know about it. So we don't need it, and the resulting code is dramatically harder to understand, plus the second arg is required (lol) so it actually gets mad at you for leaving it out. 

<img width="316" alt="image" src="https://github.com/oxidecomputer/console/assets/3612203/3393d7ef-ed0b-48a8-b708-5c44fe09347f">

All this is to say, I've already typed this so I might as well shoot it into the ether, but it is not a suggestion to use this horrible type.

Reply:
Ah, that's good to know, about, though! Thanks for writing that all out.